pkg/services/sso/keycloak_service_builder.go

```diff
@@ -0,0 +1,61 @@
+package sso
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/redhatsso"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+)
+
+var _ KeycloakServiceBuilderSelector = &keycloakServiceBuilderSelector{}
+var _ KeycloakServiceBuilder = &keycloakServiceBuilder{}
+
+type KeycloakServiceBuilder interface {
+	WithRealmConfig(realmConfig *keycloak.KeycloakRealmConfig) KeycloakServiceBuilder
+	Build() KeycloakService
+}
+
+type keycloakServiceBuilder struct {
+	config      *keycloak.KeycloakConfig
+	realmConfig *keycloak.KeycloakRealmConfig
+}
+
+func (builder *keycloakServiceBuilder) Build() KeycloakService {
+	notNilPredicate := func(x interface{}) bool {
+		return x.(*keycloak.KeycloakRealmConfig) != nil
+	}
+
+	// Temporary: if a realm configuration different from the one into the config is specified
+	// we always instantiate MAS_SSO irrespective of the selected provider
+	if builder.config.SelectSSOProvider == keycloak.MAS_SSO ||
+		builder.realmConfig != nil {
+		_, realmConfig := arrays.FindFirst(notNilPredicate, builder.realmConfig, builder.config.KafkaRealm)
```

Comment:
In this case, realm config will be `builder.config.KafkaRealm`?

Reply:
It will be `builder.config.KafkaRealm` if `builder.realmConfig` is nil, otherwise it will be `builder.realmConfig`